android/app/src/main/java/com/sample/manifest/Manifest.kt

```diff
@@ -2,14 +2,20 @@ package com.sample.manifest
 
 import com.squareup.moshi.JsonClass
 
-typealias Components = Map<String, Component>
+typealias Components = Array<Component>
 
 @JsonClass(generateAdapter = true)
 data class Manifest(
     val name: String,
     val displayName: String,
+
+    @Suppress("ArrayInDataClass")
```

Comment:
I suspect that this warning will be gone if `Array` is replaced with `List<T>` as suggested in the comment above.